core/camel-core-model/src/main/java/org/apache/camel/model/ThrottleDefinition.java

```diff
@@ -89,35 +120,132 @@ public String getShortName() {
-        return "throttle[" + getExpression() + "]";
+        return "throttle[" + description() + "]";
     }
 
     // Fluent API
     // -------------------------------------------------------------------------
     /**
-     * Sets the maximum number of concurrent requests
+     * Sets the time period during which the maximum request count is valid for
+     *
+     * @param  timePeriodMillis period in millis
+     * @return                  the builder
+     */
+    public ThrottleDefinition timePeriodMillis(long timePeriodMillis) {
+        return timePeriodMillis(Long.toString(timePeriodMillis));
+    }
+
+    /**
+     * Sets the time period during which the maximum request count is valid for
      *
-     * @param  maximumConcurrentRequests the maximum number of concurrent requests
+     * @param  timePeriodMillis period in millis
+     * @return                  the builder
+     */
+    public ThrottleDefinition timePeriodMillis(String timePeriodMillis) {
+        setTimePeriodMillis(timePeriodMillis);
+        return this;
+    }
+
+    /**
+     * Sets the maximum number of requests
+     *
+     * @param  maximumConcurrentRequests the maximum number of requests (according to the mode in use - either
+     *                                   concurrent or by time period)
      * @return                           the builder
      */
-    public ThrottleDefinition maximumConcurrentRequests(long maximumConcurrentRequests) {
+    public ThrottleDefinition maximumRequests(long maximumConcurrentRequests) {
         setExpression(
                 ExpressionNodeHelper.toExpressionDefinition(ExpressionBuilder.constantExpression(maximumConcurrentRequests)));
         return this;
     }
 
     /**
-     * Sets the number of concurrent requests
+     * Sets the maximum number of requests
      *
-     * @param  maximumConcurrentRequests the maximum number of concurrent requests
+     * @param  maximumConcurrentRequests the maximum number of requests (according to the mode in use - either
+     *                                   concurrent or by time period)
      * @return                           the builder
      */
-    public ThrottleDefinition maximumConcurrentRequests(String maximumConcurrentRequests) {
+    public ThrottleDefinition maximumRequests(String maximumConcurrentRequests) {
         setExpression(
-                ExpressionNodeHelper.toExpressionDefinition(ExpressionBuilder.simpleExpression(maximumConcurrentRequests)));
+                ExpressionNodeHelper.toExpressionDefinition(ExpressionBuilder.constantExpression(maximumConcurrentRequests)));
         return this;
     }
 
+    /**
+     * Sets the maximum number of concurrent requests
+     *
+     * @param      maximumConcurrentRequests the maximum number of concurrent requests
+     * @deprecated                           Use {@link #maximumRequests(long)}
+     * @return                               the builder
+     */
+    @Deprecated(since = "4.4.0")
```

Comment:
The new throttler code was added in 4.3 (non LTS) so IMHO its fine to remove/change the code as upgrade from LTS to LTS is what matters.

Reply:
Great. I was trying not to be super disruptive, but it makes sense. It's even better if we can get rid of this.